compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -1319,19 +1321,24 @@ trait Implicits:
        *                  return new result with preferGeneral = true
        *  3.6 and higher: compare with preferGeneral = true
        *
+       *  @param disambiguate      The call is used to disambiguate two successes, not for ranking.
+       *                           When ranking, we are always filtering out either > 0 or <= 0 results.
+       *                           In each case a priority change from 0 to -1 or vice versa makes no difference.
        */
-      def compareAlternatives(alt1: RefAndLevel, alt2: RefAndLevel): Int =
+      def compareAlternatives(alt1: RefAndLevel, alt2: RefAndLevel, disambiguate: Boolean = false): Int =
         def comp(using Context) = explore(compare(alt1.ref, alt2.ref, preferGeneral = true))
-        def warn(msg: Message) =
-          priorityChangeWarnings += ((alt1.ref, alt2.ref, msg))
         if alt1.ref eq alt2.ref then 0
         else if alt1.level != alt2.level then alt1.level - alt2.level
         else
           var cmp = comp(using searchContext())
           val sv = Feature.sourceVersion
-          if sv.stable == SourceVersion.`3.5` || sv == SourceVersion.`3.6-migration` then
+          if isWarnPriorityChangeVersion(sv) then
```

Comment:
I suppose `sv` is constant and `isWarnPriorityChangeVersion` could then be a val

Reply:
No, I use it below as well in the assert that we originally commented out.

Reply:
Oh, I meant even `isWarnPriorityChangeVersion(Feature.sourceVersion)` would be constant between calls of `compareAlternatives`. But no big deal either way, this is still ready to merge for me

Reply:
Yes, but be need sourceVersion as `sv` separately in compareAlternatives.